### PR TITLE
Test10101

### PR DESCRIPTION
**Pull request by mikelis135**
Integration Test

### CODE REVIEW

File: raveandroid/src/main/java/com/flutterwave/raveandroid/Encrypt.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.flutterwave.raveandroid;
+
+import javax.inject.Inject;
+
+public class Encrypt {
```
Comment (biodunalfet) [inline raveandroid/src/main/java/com/flutterwave/raveandroid/Encrypt.java]: @mikelis135 Delete this class. It looks unused

Comment (mikelis135) [reply raveandroid/src/main/java/com/flutterwave/raveandroid/Encrypt.java]: done

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+        ArgumentCaptor<Callbacks.OnRequeryRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnRequeryRequestComplete.class);
+        verify(networkRequest).requeryTx(any(RequeryRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onError(generateRandomString(), generateRandomString());
+        verify(view).onPaymentFailed(anyString(), anyString());
+
+    }
+
+
+    @Test
+    public void init() {
```
[the remark this answers is not in the document]
Comment (biodunalfet) [reply raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 you still haven't deleted this.

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth(AVS_VBVSECURECODE), generateRandomString());
+
+        verify(view).showProgressIndicator(false);
+        verify(view).onAVS_VBVSECURECODEModelSuggested(any(Payload.class));
+
+    }
+
+    @Test
+    public void chargeCard_onError_onPinAuthModelSuggested_showonAVS_VBVSECURECODEModelSuggested() {
+
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 the name of your test does not correspond to the test you wrote here

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+        presenter.chargeCard(generatePayload(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onError(generateRandomString(), generateRandomString());
+        verify(view).onPaymentError(anyString());
+
+    }
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onPinAuthModelSuggested_showonAVS_VBVSECURECODEModelSuggested() {
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 the name of your test does not correspond to the test you wrote here

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 5]
```diff
@@ -0,0 +1,510 @@
...
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth(PIN), generateRandomString());
+
+        verify(view).showProgressIndicator(false);
+        verify(view).onPinAuthModelSuggested(any(Payload.class));
+
+    }
+
+    @Test
+    public void chargeCard_onSuccess_onPinAuthModelSuggested_showonAVS_VBVSECURECODEModelSuggested() {
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 the name of this test is inappropriate

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java (hunk 3 above)
Comment (biodunalfet) [reply raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 and the name of the test does not make sense

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 6]
```diff
@@ -0,0 +1,510 @@
...
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth(AVS_VBVSECURECODE), generateRandomString());
+        verify(view).onAVSVBVSecureCodeModelUsed(anyString(), anyString());
+
+    }
+
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onAVS_VBVSECURECODESuggested_showAVS_VBVSECURECODES() {
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 what is the difference between this test and the one above?

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 7]
```diff
@@ -0,0 +1,510 @@
...
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateRandomChargeResponse(), generateRandomString());
+        verify(view).onPaymentError(unknownResCodemsg);
+
+    }
+
+    @Test
+    public void chargeCardWithSuggestedAuthModel_onSuccess_onUnknownAuthModelUsed_showUnknownAuthModelUsed() {
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 what does "show unknown auth model used" mean? Rename this test accordingly

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 8]
```diff
@@ -0,0 +1,510 @@
...
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth("unknown Auth"), generateRandomString());
+        verify(view).onPaymentError(unknownAuthmsg);
+
+    }
+
+    @Test
+    public void chargeCard_onError_unknownResponse() {
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 the name of this test does not describe what you're testing?

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 9]
```diff
@@ -0,0 +1,510 @@
...
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateNullChargeResponse(), generateRandomString());
+        verify(view).onPaymentError(invalidChargeCode);
+
+    }
+
+    @Test
+    public void chargeCardWithOTP_onValidateSuccessful_onOTPSuggested_showOTP() {
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 this test is confusing. Why does the name say "chargeCardWithOTP" and the first line of the test say "chargeCardWithAVSModel"?

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 10]
```diff
@@ -0,0 +1,510 @@
...
+        verify(view).showProgressIndicator(true);
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseNoAuth(), generateRandomString());
+        verify(view).onChargeCardSuccessful(any(ChargeResponse.class));
+
+    }
+
+    @Test
+    public void chargeCardWithOTP_onVaidateError_onOTPSuggested_showOTP() {
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 same feedback as the above test

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 11]
```diff
@@ -0,0 +1,510 @@
...
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseNoAuth(), generateRandomString());
+        verify(view).onChargeCardSuccessful(any(ChargeResponse.class));
+
+    }
+
+    @Test
+    public void chargeCardWithAVSModel_onSuccess_onAVS_VBVSECURECODESuggested_showAVS_VBVSECURECODES() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 same issue as above

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 12]
```diff
@@ -0,0 +1,510 @@
...
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateRandomChargeResponse(), generateRandomString());
+        verify(view).onPaymentError(unknownResCodemsg);
+
+    }
+
+    @Test
+    public void chargeCardWithAVSModel_onSuccess_onUnknownAuthModelUsed_showUnknownAuthModelUsed() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 same issue as above

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 13]
```diff
@@ -0,0 +1,510 @@
...
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateValidChargeResponseWithAuth("unknown Auth"), generateRandomString());
+        verify(view).onPaymentError(unknownAuthmsg);
+
+    }
+
+    @Test
+    public void chargeCardWithAVSModel_chargeCard_onError_unknownResponse() {
+
+        presenter.chargeCardWithSuggestedAuthModel(generatePayload(), generateRandomString(), generateRandomString(), generateRandomString());
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 same issue as above

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 14]
```diff
@@ -0,0 +1,510 @@
...
+        ArgumentCaptor<Callbacks.OnChargeRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnChargeRequestComplete.class);
+        verify(networkRequest).chargeCard(any(ChargeRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateNullChargeResponse(), generateRandomString());
+        verify(view).onPaymentError(invalidChargeCode);
+
+    }
+
+    @Test
+    public void requeryTx_onQuerySuccessful_returnOnPaymentSuccessful() {
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 rename this test. What does "returnOnPaymentSuccessful" mean?

File: raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java [hunk 15]
```diff
@@ -0,0 +1,510 @@
...
+        ArgumentCaptor<Callbacks.OnRequeryRequestComplete> captor = ArgumentCaptor.forClass(Callbacks.OnRequeryRequestComplete.class);
+        verify(networkRequest).requeryTx(any(RequeryRequestBody.class), captor.capture());
+        captor.getAllValues().get(0).onSuccess(generateRequerySuccessful(), generateRandomString());
+        verify(view).onRequerySuccessful(any(RequeryResponse.class), anyString(), anyString());
+
+    }
+
+
+    @Test
+    public void requeryTx_onQueryError_returnOnPaymentSuccessful() {
```
Comment (biodunalfet) [inline raveandroid/src/test/java/com/flutterwave/raveandroid/card/CardPresenterTest.java]: @mikelis135 same issue as above

**Pull request merged**